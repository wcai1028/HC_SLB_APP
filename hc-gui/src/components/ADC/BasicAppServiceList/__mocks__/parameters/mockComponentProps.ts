
export default (isLoading: boolean = false) => {
  if (isLoading) {
    return {
      data: new Array,
      isLoading: true,
    }
  } else {
    return {
      data: [
        {
          'app-svc': 'appSvc1',
          vport: '80+tcp',
          'app-svc-group': 'appSvcGrp1',
        },
        {
          'app-svc': 'appSvc2',
          vport: '81+udp',
          'app-svc-group': 'appSvcGrp2',
        },
      ],
      isLoading: false,
    }
  }

}
