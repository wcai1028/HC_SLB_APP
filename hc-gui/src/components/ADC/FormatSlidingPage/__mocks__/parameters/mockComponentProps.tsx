import * as React from 'react'

const minProps = {
  isOpen: true,
  children: <p>Inner Component</p>,
}

const commonProps = {
  isOpen: true,
  title: 'Common Unit Test',
  description: 'This is a common Unit Test',
  children: <p>Inner Component</p>,
  showBack: false,
  saveText: null,
  disableSave: false,
  hideCancel: false,
  customizedButtons: null,
}

const specialProps = {
  isOpen: true,
  title: 'Special Unit Test',
  description: 'This is a special Unit Test',
  children: <p>Inner Component</p>,
  showBack: true,
  saveText: 'TestSave',
  disableSave: true,
  hideCancel: true,
  customizedButtons: <button>Customized</button>,
}

export default (isSpecial?: boolean, callbackOK?: any, callbackCancel?: any) => {
  if (isSpecial === undefined && callbackOK === undefined && callbackCancel === undefined) {
    return minProps
  }

  if (isSpecial) {
    return {
      ...specialProps,
      onRequestOk: callbackOK,
      onRequestClose: callbackCancel,
    }
  } else {
    return {
      ...commonProps,
      onRequestOk: callbackOK,
      onRequestClose: callbackCancel,
    }
  }
}
