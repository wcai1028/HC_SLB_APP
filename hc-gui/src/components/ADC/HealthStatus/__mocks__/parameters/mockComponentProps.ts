export default {
  headerProps: {
    type: 'header',
  },
  ongoingProps: {
    type: 'ongoing',
  },
  lowwarnProps: {
    type: 'lowwarn',
  },
  warningProps: {
    type: 'warning',
  },
  stoppedProps: {
    type: 'stopped',
  },
  infoProps: {
    text: 'test info',
    type: 'info',
    tooltip: 'test tooltip',
  },
  unusedProps: {
    type: 'ununsed',
    default: 'test default'
  },
  numberProps: {
    type: 'number',
    size: 32,
  },
  undefinedProps: {
  },
}
