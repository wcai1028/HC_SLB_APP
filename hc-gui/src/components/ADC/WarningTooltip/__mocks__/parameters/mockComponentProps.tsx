import * as React from 'react'

interface ITestDivProps {
  onClick: () => void,
}

class TestDiv extends React.Component<ITestDivProps> {

  handleClick = () => {
    const { onClick } = this.props
    if (onClick) {
      onClick()
    }
  }

  render() {
    return <div onClick={this.handleClick}>{this.props.children}</div>
  }
}

export const mockWarningTooltipProps =  (useDefault: boolean, onConfirm: any, onVisibleChange: any) => {
  if (useDefault) {
    return {
      title: '',
      description: [],
      action: '',
      onConfirm,
      content: null,
      arrowPointAtCenter: null,
      placement: null,
      trigger: null,
      visible: undefined,
      onVisibleChange: null,
    }
  } else {
    return {
      title: 'Test Title',
      description: ['Test description1', 'Test descriptions2'],
      action: 'Save',
      onConfirm,
      content: (
        title: string,
        description: string[],
        action: string,
        onConfirm: () => void,
      ) => {
        return <>
          <div>{title}</div>
          <div>
            {description.map((data: string, index: number) => {
              return <span key={index}>{data}</span>
            })}
          </div>
          <TestDiv onClick={onConfirm} >{action}</TestDiv>
        </>
      },
      arrowPointAtCenter: false,
      placement: 'bottomRight',
      trigger: 'click',
      visible: false,
      onVisibleChange,
    }
  }
}

export const mockInfoTooltipProps = (isEmpty: boolean, isDefault: boolean, isTest: boolean) => {
  if (isEmpty) {
    return {}
  } else if (isDefault) {
    return {
      objList: [
        { value: 'Test value1' },
        { value: 'Test value2' },
        { value: 'Test value3' },
      ],
    }
  } else {
    if (isTest === true) {
      return {
        objList: [
          { value: 'Test value1' },
          { value: 'Test value2' },
          { value: 'Test value3' },
        ],
        componentDisplay: null,
        textDisplay: 'Test text',
        renderText: (obj: IObject) => {
          return <div>{obj.value}</div>
        },
        actions: 'Test action',
      }
    } else if (isTest === false) {
      return {
        objList: [
          { value: 'Test value1' },
          { value: 'Test value2' },
          { value: 'Test value3' },
        ],
        componentDisplay: <div>Test component</div>,
        textDisplay: '',
        renderText: (obj: IObject) => {
          return <div>{obj.value}</div>
        },
        actions: 'Test action',
      }
    }
    return {}
  }
}
