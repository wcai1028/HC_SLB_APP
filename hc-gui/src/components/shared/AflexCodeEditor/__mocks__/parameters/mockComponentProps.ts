
export default (onAFlexChange: any, hasDiff: boolean = false, isBeforeStyle: boolean = false) => {
  if (hasDiff) {
    return {
      onAFlexChange,
      value: 'test-value',
      options: {
        readOnly: true,
      },
      diff: [
        ['replace', 1, 3, 5, 7],
        ['insert', 1, 3, 5, 7],
        ['delete', 1, 3, 5, 7],
        ['others', 1, 3, 5, 7],
      ],
      version: isBeforeStyle ? 'before' : 'after',
      id: '123456',
    }
  } else {
    return {
      onAFlexChange,
      value: 'test-value',
      options: {
        readOnly: true,
      },
      diff: null,
      version: 'unit_test',
      id: '123456',
    }
  }

}
