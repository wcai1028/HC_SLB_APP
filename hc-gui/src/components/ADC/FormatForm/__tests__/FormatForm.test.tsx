import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'

import mockComponentProps from '../__mocks__/parameters/mockComponentProps'

const renderFormatForm = (props: any) => {
  const HCProvider = require('src/HCProvider').default
  const FormatForm = require('../index').default

  return ReactTestRenderer.create(
    <HCProvider>
      <FormatForm {...props} />
    </HCProvider>
  )
}

describe('Unit Test: FormatForm ', () => {

  describe('should render properly: ', () => {
    test('should render properly with min props', (done: any) => {
      const props = mockComponentProps()
      const testComponent = renderFormatForm(props)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with common props', (done: any) => {
      const mockCallbackOK = jest.fn()
      const mockCallbackCancel = jest.fn()
      const props = mockComponentProps(false, mockCallbackOK, mockCallbackCancel)
      const testComponent = renderFormatForm(props)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with special props', (done: any) => {
      const mockCallbackOK = jest.fn()
      const mockCallbackCancel = jest.fn()
      const props = mockComponentProps(true, mockCallbackOK, mockCallbackCancel)
      const testComponent = renderFormatForm(props)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

  })

  describe('should work properly: ', () => {
    test('should use default callback functions when provided with min props', (done: any) => {
      const props = mockComponentProps()
      const testComponent = renderFormatForm(props)

      const form = testComponent.root.find(
        (node: any) => (node.type as any).name === 'FormatForm',
      )
      form.instance.onClickSaveButton()
      form.instance.onClickCancelButton()
      form.instance.onClickBack()

      setImmediate(() => {
        done()
      })
    })

    test('should use callbacks properly when provided with common props', (done: any) => {
      const mockCallbackOK = jest.fn()
      const mockCallbackCancel = jest.fn()
      const props = mockComponentProps(false, mockCallbackOK, mockCallbackCancel)
      const testComponent = renderFormatForm(props)

      const form = testComponent.root.find(
        (node: any) => (node.type as any).name === 'FormatForm',
      )
      form.instance.onClickSaveButton()
      form.instance.onClickCancelButton()
      form.instance.onClickBack()

      setImmediate(() => {
        expect(mockCallbackOK.mock.calls.length).toBe(1)
        expect(mockCallbackCancel.mock.calls.length).toBe(2)
        done()
      })
    })

    test('should use callbacks properly when provided with special props', (done: any) => {
      const mockCallbackOK = jest.fn()
      const mockCallbackCancel = jest.fn()
      const props = mockComponentProps(true, mockCallbackOK, mockCallbackCancel)
      const testComponent = renderFormatForm(props)

      const form = testComponent.root.find(
        (node: any) => (node.type as any).name === 'FormatForm',
      )
      form.instance.onClickSaveButton()
      form.instance.onClickCancelButton()
      form.instance.onClickBack()

      setImmediate(() => {
        expect(mockCallbackOK.mock.calls.length).toBe(1)
        expect(mockCallbackCancel.mock.calls.length).toBe(2)
        done()
      })
    })
  })

})
