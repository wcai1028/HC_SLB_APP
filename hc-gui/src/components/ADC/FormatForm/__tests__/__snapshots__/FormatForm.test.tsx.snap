// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test: FormatForm  should render properly:  should render properly with common props 1`] = `
Array [
  <div
    className="a10SlidingPageContent"
  >
    <div
      className="a10SlidingPageTitle"
    >
      Common Unit Test
    </div>
    <div
      className="a10SlidingPageDescription"
    >
      This is a common Unit Test
    </div>
    <p>
      Inner Component
    </p>
  </div>,
  <div
    className="a10SlidingPageFooter"
  >
    <button
      className="ant-btn submit-button ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Save
      </span>
    </button>
    <button
      className="ant-btn cancel-button"
      onClick={[Function]}
      type="button"
    >
      <span>
        Cancel
      </span>
    </button>
  </div>,
]
`;

exports[`Unit Test: FormatForm  should render properly:  should render properly with min props 1`] = `
Array [
  <div
    className="a10SlidingPageContent"
  >
    <div
      className="a10SlidingPageTitle"
    />
    <div
      className="a10SlidingPageDescription"
    />
    <p>
      Inner Component
    </p>
  </div>,
  <div
    className="a10SlidingPageFooter"
  >
    <button
      className="ant-btn submit-button ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        Save
      </span>
    </button>
    <button
      className="ant-btn cancel-button"
      onClick={[Function]}
      type="button"
    >
      <span>
        Cancel
      </span>
    </button>
  </div>,
]
`;

exports[`Unit Test: FormatForm  should render properly:  should render properly with special props 1`] = `
Array [
  <div
    className="a10SlidingPageContent"
  >
    <div
      className="a10SlidingPageTitle"
    >
      Special Unit Test
    </div>
    <div
      className="a10SlidingPageDescription"
    >
      This is a special Unit Test
    </div>
    <p>
      Inner Component
    </p>
  </div>,
  <div
    className="a10SlidingPageFooter"
  >
    <a
      className="back-link"
      onClick={[Function]}
    >
      <i
        className="anticon anticon-double-left"
        style={Object {}}
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="double-left"
          fill="currentColor"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M272.9 512l265.4-339.1c4.1-5.2.4-12.9-6.3-12.9h-77.3c-4.9 0-9.6 2.3-12.6 6.1L186.8 492.3a31.99 31.99 0 0 0 0 39.5l255.3 326.1c3 3.9 7.7 6.1 12.6 6.1H532c6.7 0 10.4-7.7 6.3-12.9L272.9 512zm304 0l265.4-339.1c4.1-5.2.4-12.9-6.3-12.9h-77.3c-4.9 0-9.6 2.3-12.6 6.1L490.8 492.3a31.99 31.99 0 0 0 0 39.5l255.3 326.1c3 3.9 7.7 6.1 12.6 6.1H836c6.7 0 10.4-7.7 6.3-12.9L576.9 512z"
          />
        </svg>
      </i>
       Back
    </a>
    <button
      className="ant-btn submit-button ant-btn-primary"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <span>
        TestSave
      </span>
    </button>
    <button>
      Customized
    </button>
  </div>,
]
`;
